src/game/AI/ScriptDevAI/scripts/northrend/grizzly_hills.cpp

```diff
@@ -753,6 +753,34 @@ struct AttentionSpan : public AuraScript
     }
 };
 
+// 54432 - Sparksocket AA: Aggro Check
+struct SparksocketAAAggroCheck : public SpellScript
+{
+    enum
+    {
+        SPELL_SPARKSOCKETPRESP =   56320,  // Sparksocket AA: Positive Response
+    };
+
+    void OnEffectExecute(Spell* spell, SpellEffectIndex effIdx) const override
+    {
+        if (effIdx != EFFECT_INDEX_0)
+            return;
+
+        Unit* caster = spell->GetCaster();
+        Unit* target = spell->GetUnitTarget();
+
+        Player* player = (Player*)target;
```

Comment:
Do this AFTER the check if target is player

And change the C style cast to static_cast<Player*>(target);